packages/ios_platform_images/ios/Classes/UIImageIosPlatformImages.swift

```diff
@@ -0,0 +1,33 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Flutter
+import Foundation
+import UIKit
+
+extension UIImage {
+  static func flutterImage(withName name: String) -> UIImage? {
+    let imageName = (name as NSString).lastPathComponent
+    let path = (name as NSString).deletingLastPathComponent
+    let screenScale = UIScreen.main.scale
+
+    var scaledImage: UIImage?
+
+    // Search for a scaled image in the bundle
+    // The image name may be suffixed with a scale factor, e.g. @2x
+    // If a scaled image is not found, search for standard image
+
+    for scale in stride(from: Int(screenScale), through: 1, by: -1) {
```

Comment:
Why is `screenScale` being initially created with the wrong type and then cast here? (Relatedly, why is it a separate variable at all, instead of inline like before?)